src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,262 @@
+    type: "text" | "number" | "date" | "time" | string;
+    name: string;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<Row[]>,
+        SettableBoxProps,
+        SettableTableProps {
+    cols: Column[];
+    rows: Row[];
+    editTooltip: string;
+    saveTooltip: string;
+    footerLabel?: string;
+    onMouseEnter?: (row: Row) => void;
+    onMouseLeave?: (row: Row) => void;
+    rowCalculation?: (data: Row[], row: Row) => void;
+}
+
+//Mimic MUI header style for footer
+const TableCellStyleOverrides: TableCellProps["sx"] = {
+    fontFamily: "var(--defaultFont)",
+    lineHeight: "2.4rem",
+    display: "table-cell",
+    verticalAlign: "inherit",
+    textAlign: "left",
+    overflowWrap: "break-word",
+    backgroundColor: "var(--secondaryBackground)",
+    fontWeight: "bold",
+    fontSize: "1.4rem",
+    color: "var(--secondaryForeground)",
+};
+
+const calculateFooterRow = (cols: Column[], rows: Row[]): FooterRow => {
+    const totalCols = cols.filter((x) => x.columnCalculation);
+    const total: FooterRow = {};
+    for (const col of totalCols) {
+        if (col.columnCalculation) {
+            const values = rows.map((x) => x[col.name]);
+            if (col.columnCalculation === "sum") {
+                total[col.name] = calculateSum(values);
+            } else if (col.columnCalculation === "average") {
+                total[col.name] = calculateAverage(values);
+            } else {
+                total[col.name] = col.columnCalculation(values);
+            }
+        }
+    }
+    return total;
+};
+
+const calculateSum = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    return numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+};
+
+const calculateAverage = (values): number => {
+    const numericVals = values.filter((x) => typeof x == "number");
+    const sum = numericVals.length > 0 ? numericVals.reduce((s: number, a: number) => s + a, 0) : 0;
+    return sum / numericVals.length || 0;
+};
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows editing.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        rowCalculation,
+        raiseEvent,
+        setValue,
+        size,
+        stickyHeader,
+        footerLabel,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const illegalCol = cols.some((x) => ["text", "number", "date", "time"].indexOf(x.type) == -1);
+    if (illegalCol) {
+        throw new Error("Unsupported column.  Column must be of type: text, number, date or time.");
```

Comment:
This crashes the whole workflow. It is good tat we are validating it, but we can probably just log it and ensure that only the allowed types are passes to the input.

Also we should allow for `undefined`.

Also use `===` for all comparisons unless you are specifically trying to allow for type conversion.

Reply:
fixed